### PR TITLE
Sapmachine #492: Various improvements to vitals

### LINKED ISSUE #492: Various improvements to vitals
This is an umbrella issue for a number of improvements to the *vitals* feature:

1) consistently name the "comma-separated-value" output feature "csv" (it was erroneously named "cvs" in the initial commit).

2) Make it possible to reverse the printing order in the historic tables. Now, the default is we print the samples youngest-to-oldest. That is inconvenient, especially for processing in csv form. Add an option to VM.vitals to optionally reverse that order.

3) When printing comma-separated-values, enclose them in quotes. That prevents localization issues if LC_NUMERIC is not en_US.

4) The "scale" option of VM.vitals should accept bytes, aka 1, to get precise values for automated processing. This would also be consistent with what NMT does.

5) A new option should be added, "-XX:[+-]DumpVitalsAtExit" (off by default) which should dump the vitals history both in textual and csv form when the VM exits.

6) The existing "raw" output option for VM.vitals is currently only enabled on debug builds. However, it is useful in general and I would like to have it in release builds as well.

(Note: all this is supposed to be backported to our older releases).

### PR DESCRIPTION
**Pull request by tstuefe**
Various improvements to vitals.

Split into different commits, each one fixing a separate point, for ease of review. Will squash to one commit.

fixes #492

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

File: src/hotspot/share/runtime/java.cpp [hunk 1]
```diff
@@ -72,6 +72,7 @@
 #include "runtime/timer.hpp"
 #include "runtime/vmOperations.hpp"
 #include "services/memTracker.hpp"
+#include "services/stathist.hpp"
```
Comment (RealCLanger) [inline src/hotspot/share/runtime/java.cpp]: Should be prepended by a // SapMachine 2019-09-01: vitals. comment

Comment (tstuefe) [reply src/hotspot/share/runtime/java.cpp]: Thank you Christoph.

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

File: src/hotspot/share/services/stathist.cpp [hunk 2]
```diff
@@ -1170,4 +1264,32 @@ void print_report(outputStream* st, const print_info_t* pi) {
 
 }
 
+// Dump both textual and csv style reports to two files, "vitals_<pid>.txt" and "vitals_<pid>.csv".
```
[the remark this answers is not in the document]
Comment (tstuefe) [reply src/hotspot/share/services/stathist.cpp]: Klar :)

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Pull request merged**